dev/README.md

```diff
@@ -0,0 +1,666 @@
+$ pwd
+~/mountpoint-s3-csi-driver
+
+$ eksctl create cluster -f dev/mp-dev-cluster.yaml
+2025-08-20 16:40:21 [ℹ]  eksctl version 0.208.0
+2025-08-20 16:40:21 [ℹ]  using region eu-north-1
+2025-08-20 16:40:22 [ℹ]  setting availability zones to [eu-north-1b eu-north-1c eu-north-1a]
+...
+2025-08-20 16:57:04 [ℹ]  kubectl command should work with "/Users/burakvar/.kube/config", try 'kubectl get nodes'
+2025-08-20 16:57:04 [✔]  EKS cluster "mp-dev-cluster" in "eu-north-1" region is ready
+```
+
+It might take around 15-20 minutes to create the cluster, at the end it should configure your kubectl to access to your cluster.
+You can verify this by running:
+```bash
+$ kubectl get no
+NAME                                           STATUS   ROLES    AGE   VERSION
+ip-192-168-26-202.eu-north-1.compute.internal   Ready    <none>   13m   v1.33.3-eks-3abbec1
+```
+
+Now you should also get [Container Insights](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContainerInsights.html) enabled for your cluster for metrics and logs.
+
+### Install the CSI Driver Helm chart from source
+
+[Helm](https://helm.sh/) is a package manager for Kubernetes and the CSI Driver provides a Helm chart for customers to consume.
+
+The Helm chart for the CSI Driver lives in `/charts/aws-mountpoint-s3-csi-driver` folder. You'd need to re-install the Helm chart if you make any changes in that folder.
+A Helm chart is basically collection of Kubernetes objects to be installed on the target Kubernetes cluster.
+
+The CSI Driver's Helm chart basically contains Kubernetes objects for deploying the CSI Driver's Pods, namespace, permissions etc.
+
+You can follow [Helm's installation guidance](https://helm.sh/docs/intro/install/) to install it.
+
+Once you have Helm installed, you can install the CSI Driver's Helm chart from source into your Kubernetes cluster:
+```bash
+$ pwd
+~/mountpoint-s3-csi-driver
+
+$ helm upgrade --install aws-mountpoint-s3-csi-driver \
+    --namespace kube-system \
+   ./charts/aws-mountpoint-s3-csi-driver
+```
+
+This would install the Helm chart with the default values. You can customise [default values](/charts/aws-mountpoint-s3-csi-driver/values.yaml) by using `--set`:
+```bash
+$ helm upgrade --install aws-mountpoint-s3-csi-driver \
+    --namespace kube-system \
+    --set controller.nodeSelector."kubernetes\.io/role"="control-plane" \
+   ./charts/aws-mountpoint-s3-csi-driver
+```
+
+You can verify that the CSI Driver is installed by listing all Pods owned by the CSI Driver:
+```bash
+$ kubectl get pods -lapp.kubernetes.io/name=aws-mountpoint-s3-csi-driver -A
+NAMESPACE     NAME                                 READY   STATUS    RESTARTS   AGE
+kube-system   s3-csi-controller-6b4cbf9ddf-nm74l   1/1     Running   0          54s
+kube-system   s3-csi-node-dfmhd                    3/3     Running   0          54s
+```
+
+You can uninstall the Helm chart by running:
+```bash
+$ helm uninstall -n kube-system aws-mountpoint-s3-csi-driver
+```
```

Comment:
Maybe we could add a note here saying that this is unnecessary when using upgrade (since it overwrites the current).